homeassistant/components/google_air_quality/sensor.py

```diff
@@ -0,0 +1,199 @@
+    ),
+    AirQualitySensorEntityDescription(
+        key="no2",
+        translation_key="nitrogen_dioxide",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement_fn=lambda x: x.pollutants.no2.concentration.units,
+        value_fn=lambda x: x.pollutants.no2.concentration.value,
+    ),
+    AirQualitySensorEntityDescription(
+        key="o3",
+        translation_key="ozone",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement_fn=lambda x: x.pollutants.o3.concentration.units,
+        value_fn=lambda x: x.pollutants.o3.concentration.value,
+    ),
+    AirQualitySensorEntityDescription(
+        key="pm10",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.PM10,
+        native_unit_of_measurement_fn=lambda x: x.pollutants.pm10.concentration.units,
+        value_fn=lambda x: x.pollutants.pm10.concentration.value,
+    ),
+    AirQualitySensorEntityDescription(
+        key="pm25",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.PM25,
+        native_unit_of_measurement_fn=lambda x: x.pollutants.pm25.concentration.units,
+        value_fn=lambda x: x.pollutants.pm25.concentration.value,
+    ),
+    AirQualitySensorEntityDescription(
+        key="so2",
+        translation_key="sulphur_dioxide",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement_fn=lambda x: x.pollutants.so2.concentration.units,
+        value_fn=lambda x: x.pollutants.so2.concentration.value,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: GoogleAirQualityConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up sensor platform."""
+    coordinators = entry.runtime_data
+
+    for subentry_id, subentry in entry.subentries.items():
+        coordinator = coordinators[subentry_id]
+        async_add_entities(
+            (
+                AirQualitySensorEntity(coordinator, description, subentry_id, subentry)
+                for description in AIR_QUALITY_SENSOR_TYPES
+                if description.exists_fn(coordinator.data)
+            ),
+            config_subentry_id=subentry_id,
+        )
+
+
+class AirQualitySensorEntity(
+    CoordinatorEntity[GoogleAirQualityUpdateCoordinator], SensorEntity
+):
+    """Defining the Air Quality Sensors with AirQualitySensorEntityDescription."""
+
+    entity_description: AirQualitySensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: GoogleAirQualityUpdateCoordinator,
+        description: AirQualitySensorEntityDescription,
+        subentry_id: str,
+        subentry: ConfigSubentry,
+    ) -> None:
+        """Set up Air Quality Sensors."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        name = f"{subentry.data[CONF_LATITUDE]}_{subentry.data[CONF_LONGITUDE]}"
```

Comment:
Should we give the entities a nicer name, like the street name?
The problem would be, that we than could have duplicate names, if to locations in the same road.

Reply:
here: https://github.com/home-assistant/core/pull/145237/commits/c5349572eea9cb1d1ac584d32f664bace984c4ef

Reply:
It's not configureable. It's the reverse geocoding of the coordinates. So in my eyes it's userfriendly. Otherwise we just have something like this, which doesn't give much information:
<img width="416" alt="image" src="https://github.com/user-attachments/assets/acdcd045-239f-4630-bae9-728fbbf0e2e9" />

Reply:
Should this be changes to `ATTR_NAME`?